lm_eval/models/gpt2.py

```diff
@@ -29,56 +29,92 @@ def __init__(
         subfolder=None,
         tokenizer=None,
         batch_size=1,
-	max_length=None,
+        max_length=None,
         load_in_8bit: Optional[bool] = False,
         trust_remote_code: Optional[bool] = False,
         dtype: Optional[Union[str, torch.dtype]]="auto",
     ):
         super().__init__()
+        
+
+        # Initialize model
+        if isinstance(pretrained, transformers.PreTrainedModel):
```

Comment:
minor nit: it'd be nice if we could confirm this is of type `AutoModelForCausalLM` or related subclasses, since this LM subclass only assumes a causal decoder-only model type.